lib/Conversion/TritonGPUToLLVM/Utility.cpp

```diff
@@ -13,18 +13,34 @@ Value createConstantI32(Location loc, OpBuilder &rewriter, int32_t v) {
                                            IntegerAttr::get(i32ty, v));
 }
 
+Value createConstantF16(Location loc, OpBuilder &rewriter, float v) {
+  auto type = type::f16Ty(rewriter.getContext());
+  return rewriter.create<LLVM::ConstantOp>(loc, type,
+                                           rewriter.getF16FloatAttr(v));
+}
+
 Value createConstantF32(Location loc, OpBuilder &rewriter, float v) {
   auto type = type::f32Ty(rewriter.getContext());
   return rewriter.create<LLVM::ConstantOp>(loc, type,
                                            rewriter.getF32FloatAttr(v));
 }
 
-Value createConstantF64(Location loc, OpBuilder &rewriter, float v) {
+Value createConstantF64(Location loc, OpBuilder &rewriter, double v) {
   auto type = type::f64Ty(rewriter.getContext());
   return rewriter.create<LLVM::ConstantOp>(loc, type,
                                            rewriter.getF64FloatAttr(v));
 }
 
+Value createNaNConstant(Location loc, OpBuilder &rewriter, Type type) {
+  if (type.isF16())
+    return createConstantF16(loc, rewriter, std::nanf(""));
+  if (type.isF32())
+    return createConstantF32(loc, rewriter, std::nanf(""));
+  if (type.isF64())
+    return createConstantF64(loc, rewriter, std::nan(""));
+  llvm_unreachable("unsupported type");
+}
```

Comment:
Could you use `APFloat::getNan` instead? That's probably a bit safe than passing Nan around as floats.
You can then create a Float attribute with `build.getFloatAttr` and feed that to create<ConstantOp>

Reply:
Good tip! Thanks, will do that!